src/logic/claims/data.ts

```diff
@@ -95,162 +85,93 @@ export function useUserClaimById(claimId: string) {
   };
 }
 
-async function loadClaimsByIds(claimIds: string[]) {
-  await sleep();
-  return mockContractData
-    .filter((claimData) => claimIds.includes(claimData.claimId.toString()))
-    .reduce((acc, claimData) => {
-      const claim: Claim = {
-        claimId: claimData.claimId.toString(),
-        policyId: claimData.policyId.toString(),
-        evidence: claimData.evidence,
-        timestamp: blockTimestampToDate(claimData.timestamp),
-        claimant: claimData.claimant,
-        beneficiary: claimData.beneficiary,
-        claimedAmount: claimData.claimedAmount,
-        counterOfferAmount: claimData.counterOfferAmount ?? null,
-        resolvedAmount: claimData.resolvedAmount ?? null,
-        open: claimData.open,
-        status: ClaimStatuses[claimData.status as ClaimStatus],
-        statusUpdatedAt: blockTimestampToDate(claimData.statusUpdatedAt),
-        statusUpdatedBy: getStatusUpdatedByType(claimData.statusUpdatedBy),
-        deadline: null,
-        transactionHash: claimData.transactionHash,
-      };
+async function loadClaims(
+  contract: ClaimsManagerWithKlerosArbitrator,
+  params: { userAccount?: string; claimId?: BigNumber }
+): Promise<{ ids: string[]; byId: Record<string, Claim> }> {
+  const { userAccount = null, claimId = null } = params;
+  // Get all the static data via events
+  const [createdEvents, counterOfferEvents] = await Promise.all([
+    contract.queryFilter(contract.filters.CreatedClaim(claimId, userAccount)),
+    contract.queryFilter(contract.filters.ProposedSettlement(claimId, userAccount)),
+  ]);
+
+  if (!createdEvents.length) {
+    return { ids: [], byId: {} };
+  }
 
-      claim.deadline = calculateDeadline(claim);
-      acc[claim.claimId] = claim;
-      return acc;
-    }, {} as Record<string, Claim>);
-}
+  // Get all the dynamic data (status etc) via a single call
+  const calls = createdEvents.map((event) => {
```

Comment:
Shouldn't we chunk this?

Reply:
We don't expect users to create loads of claims (these are only the claims for the current user), so I would say probably not at this stage. Out of interest though, is it possible to load a subset of the events/logs?

Reply:
>  Out of interest though, is it possible to load a subset of the events/logs?

Logs can be indexed and you can filter them by the indexed fields.

Reply:
Haha my question wasn't super clear (I know we can filter the logs as I do that already). I was just wondering hypothetically, if we were to chunk, whether it would apply to the logs. But no worries

Reply:
I see. One way of doing that is specifying from/to which block search for the logs. But we can deal with that later...
E.g. we can just fetch all of them and once we see there are too many claims we can implement the chunking.